ORBIT/phases/install/install_phase.py

```diff
@@ -33,8 +34,68 @@ def __init__(self, weather, **kwargs):
 
         self.extract_phase_kwargs(**kwargs)
         self.initialize_environment(weather, **kwargs)
+
+        self.processes = self.build_processes(processes)
         self.availability = kwargs.get("availability", None)
 
+    def build_processes(self, input):
+        """
+
+        Parameters
+        ----------
+        input :
+        """
+
+        processes = {**default_processes, **input}
+
+        new = {}
+        for key, process in processes.items():
+            new[key] = self._validate_process(key, process)
+
+        return new
+
+    def _validate_process(self, key, process):
+        """
+
+        Parameters
+        ----------
+        key :
+        process :
+        """
+
+        required = ["name", "duration"]
+
+        missing = [k for k in required if k not in process.keys()]
+        if missing:
+            raise ValueError(
+                f"Missing value(s) for '{missing}' in process '{key}'."
+            )
+
+        constraints = process.pop("constraints", {})
+
+        new = {**process}
+        new["constraints"] = {}
+
+        for k, v in constraints.items():
```

Comment:
Constraints are validated here. Now there can be any # of constraints on a process. The options for constraint types for now are max_, min_, true, false.